pkg/segment/writer/segwriter.go

```diff
@@ -250,63 +306,185 @@ func cleanRecentlyRotatedInfo() {
+			// TODO: store the ascii in ple.allCvals to avoid recomputation
+			var asciiBytesBuf bytes.Buffer
+			switch ctype {
+			case VALTYPE_ENC_INT64[0]:
+				numType = SS_INT64
+				intVal = utils.BytesToInt64LittleEndian(ple.allCvalsTypeLen[i][1:9])
+				_, err := fmt.Fprintf(&asciiBytesBuf, "%d", intVal)
+				if err != nil {
+					return false, utils.TeeErrorf("doLogEventFilling: cannot write intVal %v: %v", intVal, err)
+				}
+			case VALTYPE_ENC_UINT64[0]:
+				numType = SS_UINT64
+				uintVal = utils.BytesToUint64LittleEndian(ple.allCvalsTypeLen[i][1:9])
+				_, err := fmt.Fprintf(&asciiBytesBuf, "%d", uintVal)
+				if err != nil {
+					return false, utils.TeeErrorf("doLogEventFilling: cannot write uintVal %v: %v", uintVal, err)
+				}
+			case VALTYPE_ENC_FLOAT64[0]:
+				numType = SS_FLOAT64
+				floatVal = utils.BytesToFloat64LittleEndian(ple.allCvalsTypeLen[i][1:9])
+				_, err := fmt.Fprintf(&asciiBytesBuf, "%f", floatVal)
+				if err != nil {
+					return false, utils.TeeErrorf("doLogEventFilling: cannot write floatVal %v: %v", floatVal, err)
+				}
+			default:
+				return false, utils.TeeErrorf("doLogEventFilling: shouldn't get here; ctype: %v", ctype)
+			}
+
+			updateRangeIndex(cname, ri.Ranges, numType, intVal, uintVal, floatVal)
+			addSegStatsNums(segstats, cname, numType, intVal, uintVal, floatVal, asciiBytesBuf.Bytes())
+			ss.updateColValueSizeInAllSeenColumns(cname, 9)
+		default:
+			return false, utils.TeeErrorf("doLogEventFilling: unknown ctype: %v", ctype)
+		}
+	}
+
+	for colName, foundCol := range ss.wipBlock.columnsInBlock {
+		if foundCol {
+			ss.wipBlock.columnsInBlock[colName] = false
+			continue
+		}
+		colWip, ok := ss.wipBlock.colWips[colName]
+		if !ok {
+			log.Errorf("doLogEventFilling: tried to backfill a column with no colWip! %v. This should not happen", colName)
+			return false, fmt.Errorf("tried to backfill a column with no colWip")
+		}
+		colWip.cstartidx = colWip.cbufidx
+		copy(colWip.cbuf[colWip.cbufidx:], VALTYPE_ENC_BACKFILL[:])
+		colWip.cbufidx += 1
+		ss.updateColValueSizeInAllSeenColumns(colName, 1)
+		// also do backfill dictEnc for this recnum
+		ss.checkAddDictEnc(colWip, VALTYPE_ENC_BACKFILL[:], ss.wipBlock.blockSummary.RecCount,
+			colWip.cbufidx-1)
+	}
+	return matchedCol, nil
 }
 
-func (segstore *SegStore) AddEntry(streamid string, rawJson []byte, ts_millis uint64,
+func (segstore *SegStore) AddEntry(streamid string,
 	indexName string, bytesReceived uint64, flush bool, signalType SIGNAL_TYPE, orgid uint64,
 	rid uint64, cnameCacheByteHashToStr map[uint64]string,
-	jsParsingStackbuf []byte) error {
+	jsParsingStackbuf []byte, pleArray []*ParsedLogEvent) error {
+
+	tsKey := config.GetTimeStampKey()
 
 	segstore.Lock.Lock()
 	defer segstore.Lock.Unlock()
 
-	if segstore.wipBlock.maxIdx+MAX_RECORD_SIZE >= WIP_SIZE ||
-		segstore.wipBlock.blockSummary.RecCount >= MAX_RECS_PER_WIP {
-		err := segstore.AppendWipToSegfile(streamid, false, false, false)
+	for _, ple := range pleArray {
```

Comment:
we should be taking the bytesReceived from each of the ple's len(rawjson) here add that up and then use it in line 473.